pkg/bmc/ilo4.go

```diff
@@ -0,0 +1,93 @@
+package bmc
+
+import (
+	"net/url"
+	"strings"
+)
+
+func init() {
+	registerFactory("ilo4", newILOAccessDetails, []string{"http", "https"})
+}
+
+func newILOAccessDetails(parsedURL *url.URL, disableCertificateVerification bool) (AccessDetails, error) {
+	return &iLOAccessDetails{
+		bmcType:                        parsedURL.Scheme,
+		portNum:                        parsedURL.Port(),
+		hostname:                       parsedURL.Hostname(),
+		disableCertificateVerification: disableCertificateVerification,
+	}, nil
+}
+
+type iLOAccessDetails struct {
+	bmcType                        string
+	portNum                        string
+	hostname                       string
+	disableCertificateVerification bool
+}
+
+func (a *iLOAccessDetails) Type() string {
+	return a.bmcType
+}
+
+// NeedsMAC returns true when the host is going to need a separate
+// port created rather than having it discovered.
+func (a *iLOAccessDetails) NeedsMAC() bool {
+	// For the inspection to work, we need a MAC address
+	// https://github.com/metal3-io/baremetal-operator/pull/284#discussion_r317579040
+	return true
+}
+
+func (a *iLOAccessDetails) Driver() string {
+	return "ilo"
+}
+
+func (a *iLOAccessDetails) DisableCertificateVerification() bool {
+	return a.disableCertificateVerification
+}
+
+// DriverInfo returns a data structure to pass as the DriverInfo
+// parameter when creating a node in Ironic. The structure is
+// pre-populated with the access information, and the caller is
+// expected to add any other information that might be needed (such as
+// the kernel and ramdisk locations).
+func (a *iLOAccessDetails) DriverInfo(bmcCreds Credentials) map[string]interface{} {
+	result := map[string]interface{}{
+		"ilo_username": bmcCreds.Username,
+		"ilo_password": bmcCreds.Password,
+		"ilo_address":  a.hostname,
+	}
+
+	if a.disableCertificateVerification {
+		result["ilo_verify_ca"] = false
```

Comment:
I don't see any evidence that this is a [supported driver option](https://docs.openstack.org/ironic/latest/admin/drivers/ilo.html#node-configuration) in Ironic?

Reply:
I have raised a patch in ironic for this.

Reply:
Do you have a link handy? Maybe avoid adding the feature here until we get it in ironic?

Reply:
@dtantsur https://review.opendev.org/#/c/743490/5. Its under review.

Reply:
@zaneb ilo_verify_ca attribute has been merged in ironic.